London 10 | Abubakar-Meigag | JS 2 Wk 4

Abubakar-Meigag opened the pull request:
<!--

The title for your pull request should be made in this format

CITY CLASS_NO - FIRST_NAME LAST_NAME - MODULE - WEEK_NO

For example,

London Class 7 - Chris Owen - HTML/CSS - Week 1

Please complete the details below this message

-->

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

Abubakar-Meigag committed:
alarm clock done

Abubakar-Meigag committed:
quotes part is done

Abubakar-Meigag committed:
mandatory done

kitko112 commented on 1-mandatory/1-alarmclock/alarmclock.js:
```diff
@@ -1,4 +1,35 @@
+displayRemaining(inputSet.value)
+let totalSeconds = inputSet.value;
+
+let interval = setInterval(() => {
```
Nice work, good use of setInterval. 
one question here, if we input non number to the input field what would happen then?

kitko112 commented on 1-mandatory/1-alarmclock/alarmclock.js:
```diff
@@ -1,4 +1,35 @@
+let seconds = totalSeconds % 60;
+
+let timeRemaining = document.querySelector("#timeRemaining");
+timeRemaining.textContent = `Time Remaining:     ${minutes}:${seconds}`;
```
Nice use of string literal to create the string.
If you want to keep the minutes and seconds always showing two digits like "01", "01", or "29" sec, you can have a look at `String.padStart()` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

kitko112 commented on 1-mandatory/2-quotegenerator/quotes.js:
```diff
@@ -17,10 +17,36 @@
 // You DO NOT need to understand how this function works.
+
+// style
+document.getElementById("main-div").setAttribute("style", 
```
inline styling is one way of styling the html. can you tell the benefit of creating an separated css file and put all the styling into it? and use the `class` attribute of html to assign the css class from css file.

kitko112 commented on 1-mandatory/2-quotegenerator/index.html:
```diff
@@ -12,6 +12,13 @@
   </head>
   <body>
     <!-- Write your HTML in here -->
+    <div id="main-div">
```
we can add a class for the div like `<div id="main-div" class="main">`
and create a css file like style.css, write the styling for the main-div there
```
main {
  "background-color: #f49807;
  ....
}
```
to reference the css file from the html file, you can check here https://www.freecodecamp.org/news/how-to-link-css-to-html/

kitko112 commented on 1-mandatory/2-quotegenerator/quotes.js:
```diff
@@ -17,10 +17,36 @@
 function pickFromArray(choices) {
-  return choices[Math.floor(Math.random() * choices.length)];
+let newQuote = Math.floor(Math.random() * ( quotes.length))
+  document.getElementById("inner-div").innerHTML = quotes[newQuote].quote;
```
Is there a way to remain `pickFromArray` unchanged? 
is it possible to pass quotes to it and retrieve a random quote?
If it is possible we can modify the html outside the function.
To maintain the code easier, we should keep the code with single purpose, for `pickFromArray` it makes sense that it only picks something from the array rather than modify the html elements.

```
function pickFromArray(choices) {
  return choices[Math.floor(Math.random() * choices.length)];
}


btn.addEventListener("click", function(){
  const randomQuote = pickFromArray(quotes);
  // document.getElementById  for quote
  // document.getElementById  for author
})
```

kitko112 commented on 1-mandatory/3-slideshow/slideshow.js:
```diff
@@ -1 +1,71 @@
+
+function AutoForwardMove(){
+autoSlidesShow = setInterval(() => {
+  forwardMove();
```
Nicely reuse the forwardMove function for the auto play function.
I would suggest every times we click autoFor and autoBack we better call `clearInterval(autoSlidesShow)` first, because we don't want the previous interval left running

Abubakar-Meigag replied:
thanks @kitko112 for your word and review l'll start work on it 

Thank you

Abubakar-Meigag committed:
fix  keep the minutes and seconds always showing two digits

Abubakar-Meigag committed:
add two separate functions

create two separate functions that perform the same task as the original function

Abubakar-Meigag committed:
Update quotes.js

Abubakar-Meigag committed:
Update quotes.js
